go/runtime/txpool/txpool.go

```diff
@@ -460,20 +466,49 @@ func (t *txPool) checkTxBatch(ctx context.Context, rr host.RichRuntime) {
 		return
 	}
 
+	// Check if node is synced.
+	var notReady bool
+	select {
+	case <-t.consensus.Synced():
+		// Node is synced.
+	default:
+		// Node is not synced.
+		notReady = true
+		t.logger.Debug("consensus is not synced yet")
+	}
+
+	if !notReady {
```

Comment:
The semantics are different though.  If you look at the bigger picture, `notReady` makes more sense :)

Reply:
Well, maybe it still could be improved. I would expect the code to look like this:
```go
// Check if we are ready.

if notReady {
    // Return, error or fix so that we will be ready.
}

// Now that we are ready, do something.
```

Reply:
Moving the consensus sync wait has simplified this part of the code and the `notReady` bool no longer exists :)